File: 199-binary-tree-right-side-view/199-binary-tree-right-side-view.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

function rightSideView(root: TreeNode | null): number[] {
    const arr = [];
    function order(node, h) {
        if(!node) return;
        arr[h] = node.val;
        order(node.left, h+1);
        order(node.right, h+1);
        
    }
    order(root, 0);
    return arr;
};

File: 199-binary-tree-right-side-view/199-binary-tree-right-side-view.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def __init__(self):
        self.dict = {}
        
    def levelOrder(self, node, l):
        if not node:
            return
        
        self.dict[l] = node.val
        self.levelOrder(node.left, l+1)
        self.levelOrder(node.right, l+1)
        
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        self.levelOrder(root, 0)
        return self.dict.values()
    